### PR TITLE
Address hive 3.x related test issues

### PR DESCRIPTION
**Pull request by marton-bod**
Addressing the issues outlined in https://github.com/klarna/HiveRunner/issues/120:

1.  InsertIntoTableIntegrationTest.insertDataIntoTablePrimitiveParsedStrings - in Hive 3, date/timestamp types are cast to the Hadoop-specific date/timestamp types, which failed since it attempted to cast a java.sql.Date to the Hadoop type. Introduced new converters to solve the issue.

2. PreV200HiveShellHiveCliEmulationTest - I think we can drop these tests, as this CLI is no longer supported. Uncommented the Ignore for now, but could also be deleted in due time.

3. Warning messages: "Caused by: java.sql.SQLSyntaxErrorException: Table/View 'TBLS' does not exist." - in Hive 3, if hive.in.test=false (which is the default value), it's expected that the metastore DB has already been initialised with tables (e.g. TBLS), therefore some initial test queries are run against them. But in our tests, these tables don't exist in derby, hence the warning messages. Setting hive.in.test=true by default in all unit tests ensures the tables are first initialized, circumventing the problem.

4. Warning messages: "WARN DataNucleus.MetaData:96 - Metadata has jdbc-type of null yet this is not valid. Ignored" - Couldn't find the exact root cause, but seems to be a commonplace issue based on some research, without breaking functionality. Added a log throttler to limit it to a single warn log per test.

### CODE REVIEW

File: src/main/java/com/klarna/hiverunner/StandaloneHiveRunner.java [hunk 1]
```diff
@@ -68,6 +68,20 @@
...
+     * Set some properties which make good defaults for test scenarios. These can be overridden
+     * in the individual HiveShell instances though, if needed.
+     */
+    private void initializeConfig() {
+        /**
+         * If hive.in.test=false (default), Hive 3 will assume that the metastore rdbms has already been initialized
+         * with some basic tables and will try to run initial test queries against them.
+         * This results in multiple warning stacktraces if the rdbms has not actually been initialized.
+         */
+        config.getHiveConfSystemOverride().put("hive.in.test", "true");
```
Comment (massdosage) [inline src/main/java/com/klarna/hiverunner/StandaloneHiveRunner.java]: I had a feeling there must be something like this to control it, fantastic!

File: src/test/java/com/klarna/hiverunner/PreV200HiveShellHiveCliEmulationTest.java [hunk 2]
```diff
@@ -29,7 +30,7 @@
 import com.klarna.hiverunner.config.HiveRunnerConfig;
 import com.klarna.hiverunner.sql.cli.hive.PreV200HiveCliEmulator;
 
-//@Ignore TODO: some of these tests fail with Hive 3, should we fix or remove this functionality?
+@Ignore("Disabling the test due to failures when upgrading to Hive 3. This CLI has been deprecated.")
```
Comment (massdosage) [inline src/test/java/com/klarna/hiverunner/PreV200HiveShellHiveCliEmulationTest.java:33]: Let's just delete this class, I don't think it has any value for Hive 3.

**Review by massdosage — CHANGES_REQUESTED** [review]
This looks really good, thanks for the updates and the detailed explanations. I think we can remove the broken test case, otherwise all the rest LGTM.

File: src/test/java/com/klarna/hiverunner/PreV200HiveShellHiveCliEmulationTest.java (hunk 2 above)
Comment (marton-bod) [reply src/test/java/com/klarna/hiverunner/PreV200HiveShellHiveCliEmulationTest.java]: Thanks for the review! 
I've deleted the test class

**Review by patduin — APPROVED** [review]
nice, thanks!

**Review by massdosage — APPROVED** [review]

**Comment by marton-bod** [comment]
Thanks for the reviews @massdosage @patduin! 
Do you have any indication when this Hive3-compatible version could be released?

**Comment by massdosage** [comment]
No problem, I just noticed we need to use `HIVE_IN_TEST.getVarname()` instead of `HIVE_IN_TEST.name()`. I'll make that change now and push it shortly.

In terms of doing a release, I'll send a mail out to the HiveRunner list suggesting a major release for Hive 3 and that this will be the default version going forward with some suggestions what we do for Hive 2 in case there are changes in future. If there are no objections we can probably get a release done next week.

**Comment by marton-bod** [comment]
Thanks for the correction. And great news regarding the release, looking forward to it 👍

**Pull request merged**